src/components/visualizations/chart/legend/HeatMapLegend.tsx

```diff
@@ -9,64 +9,65 @@ import { formatLegendLabel } from '../../utils/common';
 
 export interface IHeatMapLegendProps {
     series: any;
+    isSmall: boolean;
     format?: string;
     numericSymbols: string[];
 }
+const DEFAULT_LABEL_LENGTH = 4;
+const SMALL_LABEL_LENGTH = 3;
 
 export default class HeatMapLegend extends React.PureComponent<IHeatMapLegendProps> {
     public render() {
-        const { series, format, numericSymbols } = this.props;
+        const { series, format, numericSymbols, isSmall } = this.props;
         const min = get(head(series), 'range.from', 0);
         const max = get(last(series), 'range.to', 0);
-        const diff =  max - min;
+        const diff = max - min;
+
+        const legendLabels = range(series.length + 1).map((index) => {
+            let value;
+
+            if (index === 0) {
+                value = formatLegendLabel(get(series, '0.range.from', 0), format, diff, numericSymbols);
```

Comment:
all three paths call formatLegendLabel with same params except first. What about keep just getting the number from series and move call of formatLefendLabel behind if-else statements